### PR TITLE
Add API conversion interface for Embedding layer.

### LINKED ISSUE #5623: Request for contributions: writing and testing API conversion decorators.
In Keras 2 (branch `keras-2`), the API for pretty much every layer has changed. We will keep supporting the Keras 1 API, and we will do so via **API conversion decorators**.

As an example, I have written the conversion decorator for the `Dense` layer. It was done in this commit: https://github.com/fchollet/keras/commit/5516c8fb42f01622c9a4a31c19cb46ae68cfa0f6
The decorator code is here: https://github.com/fchollet/keras/blob/keras-2/keras/legacy/interfaces.py#L15
The test for it is here: https://github.com/fchollet/keras/blob/keras-2/tests/keras/legacy/interface_test.py#L8

In order to launch Keras 2, we will need such decorators for every layer (and also for training routines and model constructors). If you want to contribute, please add these decorators (in `keras/legacy/interfaces.py`, with tests in `tests/keras/legacy/interface_test.py`). You just need to follow the template laid out with the `Dense` layer decorator.

You can leave out the `Conv*D` layers for now, since they will require special handling due to a positional/keyword argument conflict.

Thanks a lot!

### PR DESCRIPTION
**Pull request by rilut**
Fixes #5623.

Am I still missing some tests?  
I'm still confused about the positional argument and the layer signature.

Thanks.

### CODE REVIEW

File: keras/legacy/interfaces.py [hunk 1]
```diff
@@ -74,3 +74,50 @@ def wrapper(*args, **kwargs):
...
+        func: `__init__` method of `Embedding`.
+
+    # Returns
+        A constructor conversion wrapper.
+    """
+    @six.wraps(func)
+    def wrapper(*args, **kwargs):
+        if len(args) > 1:
+            # The first entry in `args` is `self`.
+            raise TypeError('The `Embedding` layer cannot have any positional argument')
```
Comment (fchollet) [inline keras/legacy/interfaces.py]: Actually it can. The first two are positional (input_dim, output_dim).

File: tests/keras/legacy/interface_test.py [hunk 2]
```diff
@@ -27,6 +27,19 @@ def test_dense_legacy_interface():
                                    bias_constraint='max_norm', name='d')
     assert json.dumps(old_layer.get_config()) == json.dumps(new_layer.get_config())
 
+@keras_test
+def test_embedding_legacy_interface():
+    old_layer = keras.layers.Embedding(input_dim=4, output_dim=2, name='d',
+                                   init='normal',
```
Comment (fchollet) [inline tests/keras/legacy/interface_test.py]: You should use a PEP8 linter to avoid style issues. https://pypi.python.org/pypi/pep8

File: keras/legacy/interfaces.py (hunk 1 above)
Comment (rilut) [inline keras/legacy/interfaces.py]: but I think I've tried `args[1]` and it's out of bounds?

Comment (fchollet) [inline keras/legacy/interfaces.py]: `args` is the list of positional arguments passed by the user. It could have any length.

Comment (rilut) [inline keras/legacy/interfaces.py]: oh ok

File: tests/keras/legacy/interface_test.py [hunk 3]
```diff
@@ -28,5 +28,17 @@ def test_dense_legacy_interface():
     assert json.dumps(old_layer.get_config()) == json.dumps(new_layer.get_config())
 
 
+@keras_test
+def test_embedding_legacy_interface():
+    old_layer = keras.layers.Embedding(input_dim=4, output_dim=2, name='d',
```
Comment (fchollet) [inline tests/keras/legacy/interface_test.py]: You should also add tests to cover positional arguments (here you only use keyword arguments). Current code wouldn't work for positional arguments.

Comment (rilut) [inline tests/keras/legacy/interface_test.py]: old_layer = keras.layers.Embedding(4, 2, name='d')
    new_layer = keras.layers.Embedding(output_dim=2, input_dim=4, name='d')
    assert json.dumps(old_layer.get_config()) == json.dumps(new_layer.get_config())

like that? is that enough?

File: keras/legacy/interfaces.py [hunk 4]
```diff
@@ -74,3 +74,64 @@ def wrapper(*args, **kwargs):
...
+
+        args = (args[0], input_dim, output_dim)
+
+        converted = []
+
+        # Remaining kwargs.
+        conversions = [
+            ('init', 'embeddings_initializer'),
+            ('W_regularizer', 'embeddings_regularizer'),
+            ('W_constraint', 'embeddings_constraint'),
```
Comment (fchollet) [inline keras/legacy/interfaces.py]: Additionally, the old `Embedding` supported a `dropout` argument that we no longer support. Since calls using the old API should still work, the dropout argument needs to be handled. Just remove it from kwargs if present, and raise a warning saying that the argument no longer exists that people should use instead a `keras.layers.SpatialDropout1D` layer right after the `Embedding` layer to get the same behavior.

Comment (fchollet) [inline keras/legacy/interfaces.py]: Also add a test for it.

**Comment by rilut** [comment]
Hi, sorry I'm closing this PR. Too many conflicts here
